blog/content/post/2020-02-13-rpi4_yocto.md

```diff
@@ -0,0 +1,272 @@
+double-precision and a single-precision accelerated by NEON instruction
+available since Raspberry Pi 2. From a benchmark perspective, we can notice
+a tremendous increase in performance over Pi 3B+ model:
+* 4x greate CPU performance
+* 2x faster browsing speed
+* 50% more fps while playing Quake III!
+
+Among other countless changes in performance, such as: improved Memory
+bandwidth, faster GPIO switching, better file compression. First time in
+history raspberry offers customizability. Now we can choose between 1,2 and
+even 4GB of RAM
+
+## Building minimal image with Yocto
+
+Following a rule of embedded systems saying not to build things that you will
+never use. We are going to build a core-minimal-image. It's an image with extremely 
+small footprint and its a great thing to start with. 
+
+Before we start make sure that all required tools are installed. Type:
+ 
+```shell 
+sudo apt-get install gawk wget git-core diffstat unzip texinfo gcc-multilib
+build-essential chrpath socat libsdl1.2-dev xterm python
+```
+to install missing packages. 
+
+The most common tool to connect with raspberry is uart interface. It will be also
+used in this post. I also use usb type C to power up the device.
+At this point, we can start for real. First of all, let's download Yocto then
+switch to the "zeus" branch which is the newest community version of Yocto
+at the time of writing this post.
+
+```shell
+mkdir Yocto
+cd Yocto
+git clone -b zeus http://git.yoctoproject.org/git/poky
+```
+When cloning is finished we need to create a build directory and set up all the
+environmental variables, paths, etc. It can be obtained by typing
+
+```shell
+cd poky
+source oe-init-build-env
+```
+Yocto repository contains an extremally useful layer containing board support
+package called meta-raspberry pi that we need to add to our working directory
+
+```shell
+cd ~/Yocto
+git clone -b zeus git://git.yoctoproject.org/meta-raspberrypi
+```
+
+Before we build we need to make some changes in crucial configuration files
+Let's start from editing `bblayers.conf` located in `~/Yocto/poky/build`.
+This file tells bitbake about the location of various layers locations which
+are used to configure things like the target of a build and the location of
+output files. 
+
+Fist of all we need to add `meta-raspberrypi` layer we
+can do that by appending it to BBLAYERS. The resault should look like this
+
+```shell
+BBLAYERS ?= " \
+<projectPath>/Yocto/poky/meta \
+<projectPath>/Yocto/poky/meta-poky \
+<projectPath>/Yocto/poky/meta-yocto-bsp \
+<projectPath>/Yocto/meta-raspberrypi \
+" 
+```
+
+Then let's move to editing another file in the same directory. This time
+we dive into local.conf which is one of the most important files where
+many changes take place.
+
+First of all, we have to specify the machine on which we are working on.
+To do that just edit line containing `MACHINE ??=` to look like this:
+
+```shell
+MACHINE ??= "raspberrypi4-64.conf" 
```

Comment:
@pawelzaborny Are you sure?

Reply:
Ofc not. It should be without `.conf` as I remember